Engine/DataFeeds/Enumerators/Factories/BaseDataCollectionSubscriptionEnumeratorFactory.cs

```diff
@@ -53,24 +42,33 @@ public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IData
             using (var dataCacheProvider = new SingleEntryDataCacheProvider(dataProvider))
             {
                 var configuration = request.Configuration;
-                var tradableDays = _tradableDaysProvider(request);
                 var sourceFactory = (BaseData)Activator.CreateInstance(request.Configuration.Type);
 
-                // Note: this enumerator factory is currently only used in backtesting with coarse data
-                // and has been updated to behave in the same way as in live trading
+                // we want the first selection to happen on the start time
+                // so we need the previous tradable day time, since coarse
+                // files are for each tradable date but emitted with next day time
+                var previousTradableDay = Time.GetStartTimeForTradeBars(
+                    request.Security.Exchange.Hours,
+                    request.StartTimeLocal,
+                    Time.OneDay,
+                    1,
+                    false);
+                var tradableDays = new[] { previousTradableDay }.Concat(request.TradableDays);
+
+                // Behaves in the same way as in live trading
                 // (i.e. only emit coarse data on dates following a trading day)
                 // The shifting of dates is needed to ensure we never emit coarse data on the same date,
                 // because it would enable look-ahead bias.
 
                 // shift all tradeable dates forward one day
-                foreach (var date in tradableDays.Select(x => x.AddDays(1)))
+                foreach (var date in tradableDays)
```

Comment:
Just leaving note for posterity; not sure changes are needed other than the comment:
> shift all tradeable dates forward one day
>to:  shift all date of emitting the file forward one day to model emitting coarse midnight the next day.

Also just want to sanity-check need to confirm this won't change the dates the live enumerator stack processes.

-------
The original code, although messy was on purpose to model _time_ as in live mode. In live _time_ is always end time. Universe data from Monday is deployed on Tuesday at midnight, and as the BaseDataCollection has no period it deploys instantly. This for loop was intended to model this live behavior:

1) `tradeableDays.AddDays(1)` = D+1 at midnight end time set time, emitting on a Saturday for Friday's data; Tuesday for Monday. (like the real daily clock)
2) `GetSource()` for `date.AddDays(-1)` = Specifically fetch the file named Monday at Tuesday midnight,
3) `ForSource(Days-1)` = Set the time of the datapoint to Monday in the coarseFundamental.Reader( date )
4) yield return the datapoint on Tuesday midnight, the real end time of the bar.

Perhaps could think of this (and update comment) to be:
1) Enumerate time start of Monday midnight (including since before start of backtest).
2) Get Source()/factor file for Monday before backtest starts.
3) Emit for backtesting 00:00 Tuesday morning?

The new loop _time_ does not model live trading IMHO; but it's fine if it addresses this bug.

Reply:
> Also just want to sanity-check need to confirm this won't change the dates the live enumerator stack processes.

Correct 👍  , he is only used in backtesting https://github.com/QuantConnect/Lean/pull/3452/files#diff-312a6d73563b5bf4e87823d1637c8425R31 
Good catch, I'll update the comment.

Reply:
EPIC!!